src/main/java/org/folio/services/validator/registry/ValidatorRegistryServiceImpl.java

```diff
@@ -0,0 +1,204 @@
+    try {
+      CQLWrapper cql = getCQL(query, length, start - 1);
+      String[] fieldList = {"*"};
+      PostgresClient.getInstance(vertx, tenantId).get(VALIDATION_RULES_TABLE_NAME, Rule.class, fieldList, cql, true, false, getReply -> {
+        if (getReply.failed()) {
+          logger.error("Error while querying the db to get all tenant rules", getReply.cause());
+          asyncResultHandler.handle(Future.failedFuture(getReply.cause()));
+        } else {
+          RuleCollection rules = new RuleCollection();
+          List<Rule> ruleList = (List<Rule>) getReply.result().getResults();
+          rules.setRules(ruleList);
+          rules.setTotalRecords(ruleList.size());
+          asyncResultHandler.handle(Future.succeededFuture(JsonObject.mapFrom(rules)));
+        }
+      });
+    } catch (Exception e) {
+      logger.error("Error while getting all tenant rules", e);
+      asyncResultHandler.handle(Future.failedFuture(e));
+    }
+    return this;
+  }
+
+  /**
+   * Creates rule for tenant with specified id
+   *
+   * @param tenantId           tenant id
+   * @param validationRule     rule to save
+   * @param asyncResultHandler result handler
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Override
+  public ValidatorRegistryService createTenantRule(String tenantId, JsonObject validationRule, Handler<AsyncResult<JsonObject>> asyncResultHandler) {
+    try {
+      String id = UUID.randomUUID().toString();
+      validationRule.put(RULE_ID_FIELD, id);
+      PostgresClient.getInstance(vertx, tenantId).save(VALIDATION_RULES_TABLE_NAME, id, validationRule.mapTo(Rule.class), postReply -> {
+        if (postReply.failed()) {
+          logger.error("Error while saving the rule to the db", postReply.cause());
+          asyncResultHandler.handle(Future.failedFuture(postReply.cause()));
+        } else {
+          asyncResultHandler.handle(Future.succeededFuture(validationRule));
+        }
+      });
+    } catch (Exception e) {
+      logger.error("Error while creating new tenant rule", e);
+      asyncResultHandler.handle(Future.failedFuture(e));
+    }
+    return this;
+  }
+
+  /**
+   * Updates rule for tenant with specified id by identifier from given <code>validationRule</code>
+   *
+   * @param tenantId           tenant id
+   * @param validationRule     rule to update
+   * @param asyncResultHandler result handler
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Override
+  public ValidatorRegistryService updateTenantRule(String tenantId, JsonObject validationRule, Handler<AsyncResult<JsonObject>> asyncResultHandler) {
+    try {
+      String id = validationRule.getString(RULE_ID_FIELD);
+      Criteria idCrit = constructCriteria(RULE_ID_JSONB_FIELD, id);
+      PostgresClient.getInstance(vertx, tenantId).get(VALIDATION_RULES_TABLE_NAME, Rule.class, new Criterion(idCrit), true, false, getReply -> {
```

Comment:
Consider removing this get(...) request.
putReply.result().getUpdated() reports the number of rules that match idCrit.